src/txn/txn_ckpt.c

```diff
@@ -640,9 +640,8 @@ __txn_checkpoint(WT_SESSION_IMPL *session, const char *cfg[])
 	WT_ASSERT(session, session->ckpt_handle_next == 0);
 	WT_WITH_SCHEMA_LOCK(session,
 	    WT_WITH_TABLE_LOCK(session,
-		WT_WITH_HANDLE_LIST_LOCK(session,
-		    ret = __checkpoint_apply_all(
-		    session, cfg, __wt_checkpoint_get_handles, NULL))));
+		ret = __checkpoint_apply_all(
```

Comment:
@agorrod, I believe the schema lock protects against creates / drops during this walk.

Also note that there is already some ambiguity about what set of files are include in a checkpoint (if a file is opened for the first time and dirtied in between here and the transaction starting, it won't be included in the checkpoint).  I'd like to leave fixing that out of scope for this change.